homeassistant/components/tesla_fleet/__init__.py

```diff
@@ -51,6 +54,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: TeslaFleetConfigEntry) -
     scopes = token["scp"]
     region = token["ou_code"].lower()
 
+    OAuth2FlowHandler.async_register_implementation(
+        hass,
+        TeslaOAuth2Implementation(hass, DOMAIN, ClientCredential(CLIENT_ID, "", NAME)),
```

Comment:
We shouldn't use the application credentials implementation base class if we're registering a custom implementation outside of application credentials flow.